notify/impl.go

```diff
@@ -719,6 +726,50 @@ func (n *OpsGenie) Notify(ctx context.Context, as ...*types.Alert) error {
 	return nil
 }
 
+// AWSSNS implements a Notifier for AWS SNS notifications.
+type AWSSNS struct {
+	conf *config.AWSSNSConfig
+	tmpl *template.Template
+}
+
+func NewAWSSNS(c *config.AWSSNSConfig, t *template.Template) *AWSSNS {
+	return &AWSSNS{conf: c, tmpl: t}
+}
+
+func (*AWSSNS) name() string { return "awssns" }
+
+func (n *AWSSNS) Notify(ctx context.Context, as ...*types.Alert) error {
+	key, ok := GroupKey(ctx)
+	if !ok {
+		return fmt.Errorf("group key missing")
+	}
+
+	log.With("incident", key).Debugln("notifying AWS SNS")
+
+	data := n.tmpl.Data(receiver(ctx), groupLabels(ctx), as...)
+
+	var err error
+	tmpl := tmplText(n.tmpl, data, &err)
+
+	config := aws.NewConfig().WithRegion(n.conf.AWSRegion)
+	svc := sns.New(session.New(config))
+
+	subject := tmpl(n.conf.Subject)
+	params := &sns.PublishInput{
+		Message:  aws.String(tmpl(n.conf.Message)),
+		Subject:  aws.String(subject[:99]),
+		TopicArn: aws.String(tmpl(n.conf.TopicArn)),
+	}
+	_, err = svc.Publish(params)
+
+	if err != nil {
+		log.Warnf("sns publish failed: %s", err)
+		return fmt.Errorf("failed to publish sns message")
```

Comment:
sns -> SNS

Could you include the actual error message, like:

`return fmt.Errorf("failed to publish SNS message: %v", err)`

I would also remove the logging above it, because the returned error will already be logged by the caller.